### PR TITLE
Fixed #15982 -- Added DATE_INPUT_FORMATS to forms.DateTimeField default input formats.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: django/conf/global_settings.py [hunk 1]
```diff
@@ -381,15 +381,12 @@ def gettext_noop(s):
...
     '%Y-%m-%d %H:%M',        # '2006-10-25 14:30'
-    '%Y-%m-%d',              # '2006-10-25'
     '%m/%d/%Y %H:%M:%S',     # '10/25/2006 14:30:59'
     '%m/%d/%Y %H:%M:%S.%f',  # '10/25/2006 14:30:59.000200'
     '%m/%d/%Y %H:%M',        # '10/25/2006 14:30'
-    '%m/%d/%Y',              # '10/25/2006'
     '%m/%d/%y %H:%M:%S',     # '10/25/06 14:30:59'
     '%m/%d/%y %H:%M:%S.%f',  # '10/25/06 14:30:59.000200'
     '%m/%d/%y %H:%M',        # '10/25/06 14:30'
-    '%m/%d/%y',              # '10/25/06'
```
Comment (felixxm) [inline django/conf/global_settings.py]: I think we should update [DATETIME_INPUT_FORMATS docs](https://docs.djangoproject.com/en/dev/ref/settings/#std:setting-DATETIME_INPUT_FORMATS) and probably add a note to the `forms.DateTimeField.input_formats` docs.

File: django/forms/fields.py [hunk 2]
```diff
@@ -435,9 +436,18 @@ def strptime(self, value, format):
         return datetime.datetime.strptime(value, format).time()
 
 
+class DatetimeFormatsIterator:
```
Comment (felixxm) [inline django/forms/fields.py]: `DatetimeFormatsIterator` -> `DateTimeFormatIterator`.

File: django/forms/fields.py [hunk 3]
```diff
@@ -435,9 +436,18 @@ def strptime(self, value, format):
...
+    def __iter__(self):
+        for item in chain(
+            formats.get_format('DATETIME_INPUT_FORMATS'),
+            formats.get_format('DATE_INPUT_FORMATS')
+        ):
+            yield item
```
Comment (felixxm) [inline django/forms/fields.py]: `itertools.chain()` returns an iterator so I think we can simply use it in `DateTimeField.input_formats`, :thinking: :
```python
input_formats = chain(
    formats.get_format('DATETIME_INPUT_FORMATS'),
    formats.get_format('DATE_INPUT_FORMATS'),
)
```
What do you think?

Comment (claudep) [reply django/forms/fields.py]: I think I tried it and the problem was related to lazyness, as chaining lazy methods doesn't seem to work properly.

Comment (felixxm) [reply django/forms/fields.py]: But `formats.get_format()` is not lazy, am I right? :thinking:. We can also use `yield from ... ` without `chain()`, e.g.
```python
def __iter__(self):
    yield from formats.get_format('DATETIME_INPUT_FORMATS')
    yield form formats.get_format('DATE_INPUT_FORMATS')
```

Comment (claudep) [reply django/forms/fields.py]: Good idea for the double `yield from`. I will still test your direct `chain()` proposal tonight and report.

Comment (claudep) [reply django/forms/fields.py]: So, testing show that using `get_format()` at import time is not an option, and chaining the `get_format_lazy()` version is not working.

**Comment by claudep** [comment]
Thanks, your edits make sense.

**Pull request merged**